spec/models/metric_spec.rb

```diff
@@ -67,17 +67,17 @@
           expect(MiqQueue.count).to eq(47)
 
           expected_targets = Metric::Targets.capture_targets
-          expected = expected_targets.collect do |t|
+          expected = expected_targets.flat_map do |t|
             # Storage is hourly only
             # Non-storage historical is expecting 7 days back, plus partial day = 8
-            t.kind_of?(Storage) ? [t, "hourly"] : [[t, "realtime"], [t, "historical"] * 8]
-          end.flatten
+            t.kind_of?(Storage) ? [t, "hourly"] : [[t, "realtime"], [t, "historical"] * 8].flatten
+          end
 
-          selected = MiqQueue.where(:method_name => "perf_capture").order(:id).collect do |q|
+          selected = MiqQueue.where(:method_name => "perf_capture").order(:id).flat_map do |q|
```

Comment:
If you don't care about the order anymore, then you can probably drop the `order(:id)`.  Makes it clearer that order is not important in the test itself.